lib/project_types/script/layers/application/push_script.rb

```diff
@@ -9,29 +9,36 @@ def call(ctx:, force:, project:)
             script_project_repo = Infrastructure::ScriptProjectRepository.new(ctx: ctx)
             script_project = script_project_repo.get
             script_project.env = project.env
+
             task_runner = Infrastructure::Languages::TaskRunner
               .for(ctx, script_project.language, script_project.script_name)
 
-            extension_point = ExtensionPoints.get(type: script_project.extension_point_type)
-            library_name = extension_point.libraries.for(script_project.language)&.package
-            raise Infrastructure::Errors::LanguageLibraryForAPINotFoundError.new(
-              language: script_project.language,
-              api: script_project.extension_point_type
-            ) unless library_name
+            if task_runner.respond_to?(:build)
+              extension_point = ExtensionPoints.get(type: script_project.extension_point_type)
+              library_name = extension_point.libraries.for(script_project.language)&.package
+              raise Infrastructure::Errors::LanguageLibraryForAPINotFoundError.new(
+                language: script_project.language,
+                api: script_project.extension_point_type
+              ) unless library_name
+
+              library = {
+                language: script_project.language,
+                version: task_runner.library_version(library_name),
+              }
 
-            library = {
-              language: script_project.language,
-              version: task_runner.library_version(library_name),
-            }
+              ProjectDependencies.install(ctx: ctx, task_runner: task_runner)
+              BuildScript.call(ctx: ctx, task_runner: task_runner, script_project: script_project, library: library)
+            end
```

Comment:
Instead of having this conditional, could we just have the `WasmTaskRunner#build` method just read the file contents at `build/script.wasm`?

Reply:
I'm definitely not a huge fan of this conditional. `responds_to?` seems kind of dangerous, because I can imagine a version of our our where the base task_runner class is an interface and provides a `NotImplemented` version of `build` which would satisfy this check. I'm going to give this a rework today.

Reply:
I've added `build` to the `WasmTaskRunner`. This also allows us to throw our existing `WebAssemblyBinaryNotFoundError` during this step.